### PR TITLE
finalise docupdater redis migration

### PR DESCRIPTION
**Pull request by briangough**
### Description

Removes the project history migration code from docupdater.  After this is deployed docupdater will  always push onto the project history queues in the new redis.  It will not check whether it should still write to the old one.

#### Screenshots

NA

#### Related Issues / PRs

https://github.com/overleaf/issues/issues/2973

### Review

There will be a separate PR for project-history, so that it only reads from the new redis and never checks the old redis.

This partly addresses the issue in https://github.com/overleaf/document-updater/pull/66 @das7pad, but there is still further work to clean up track-changes.  Track-changes can only live in the main redis at the moment, because its operations are in a `multi` with the document updates.  This means that the separate `redis.history` setting at [settings.defaults.coffee#L30-L37](https://github.com/overleaf/document-updater/blob/a396ad17fe4ccce2b2106308959ac5956099a9e8/config/settings.defaults.coffee#L30-L37) is redundant.

#### Potential Impact

High, because it's docupdater.  If the history updates fail the history for all currently active projects will have to be resynced.

If we clean this up to remove the `NEW_` in future that will also be a dangerous change.

#### Manual Testing Performed

- [x]  Unit and acceptance tests 
- [ ] Deploy in staging while chaosmonkey is running 

#### Accessibility

NA

### Deployment

Before deploying this it's important that every docupdater is writing to the new redis and there is no project-history data in the old redis (i.e. every project has a ProjectHistory:MigrationKey set to `NEW` in the new redis).  If  these conditions are satisfied then nothing should change.

Double check the settings when this goes out because they must be correctly set to connect to the new redis for NEW_HISTORY_REDIS.    This is how they are currently.

#### Deployment Checklist

Before deployment:

- [x] Check that there is no history in the old redis with `redis-cli  -h $REDIS_HOST -p $REDIS_PORT -a $REDIS_PASSWORD  --scan --pattern ProjectHistory:Ops:*'`

After deployment

- [ ]  Remove the migration flag keys from the new redis, as they are no longer needed. `redis-cli  -h $NEW_HISTORY_REDIS_HOST -p $NEW_HISTORY_REDIS_PORT -a $NEW_HISTORY_REDIS_PASSWORD  --scan --pattern 'ProjectHistory:MigrationKey:*' > /tmp/keys.txt` 

These keys can be deleted `(while read LINE ; do echo "DEL $LINE"; done ) < /tmp/keys.txt | redis-cli ...`

#### Metrics and Monitoring

The docupdater metrics should disappear from the migration chart on the project-history dashboard.

#### Who Needs to Know?

@emcsween @henryoswald

### CODE REVIEW

File: config/settings.defaults.coffee [hunk 1]
```diff
@@ -37,23 +37,12 @@ module.exports =
...
-			host: process.env["HISTORY_REDIS_HOST"] or process.env["REDIS_HOST"] or "localhost"
-			password: process.env["HISTORY_REDIS_PASSWORD"] or process.env["REDIS_PASSWORD"] or ""
-			maxRetriesPerRequest: parseInt(process.env['REDIS_MAX_RETRIES_PER_REQUEST'] or "20")
-			key_schema:
-				projectHistoryOps: ({project_id}) -> "ProjectHistory:Ops:{#{project_id}}"
-				projectHistoryFirstOpTimestamp: ({project_id}) -> "ProjectHistory:FirstOpTimestamp:{#{project_id}}"
-
-		new_project_history:
 			port: process.env["NEW_HISTORY_REDIS_PORT"] or "6379"
 			host: process.env["NEW_HISTORY_REDIS_HOST"]
```
Comment (emcsween) [inline config/settings.defaults.coffee:41]: Should there be a fallback to REDIS_HOST for the dev environment?

File: config/settings.defaults.coffee [hunk 2]
```diff
@@ -37,23 +37,12 @@ module.exports =
 				docsWithHistoryOps: ({project_id}) -> "DocsWithHistoryOps:{#{project_id}}"
 
 		project_history:
-			port: process.env["HISTORY_REDIS_PORT"] or process.env["REDIS_PORT"] or "6379"
-			host: process.env["HISTORY_REDIS_HOST"] or process.env["REDIS_HOST"] or "localhost"
-			password: process.env["HISTORY_REDIS_PASSWORD"] or process.env["REDIS_PASSWORD"] or ""
-			maxRetriesPerRequest: parseInt(process.env['REDIS_MAX_RETRIES_PER_REQUEST'] or "20")
```
Comment (emcsween) [inline config/settings.defaults.coffee:43]: Is this removed on purpose?

Comment (briangough) [reply config/settings.defaults.coffee:43]: Ah yes, better to leave that in. I will remove the keepAlive:100 as we previously removed that in 8a46a2726e651288b9b3269873beeac30154af69.

File: config/settings.defaults.coffee (hunk 1 above)
Comment (briangough) [reply config/settings.defaults.coffee:41]: Yes, I'll add a fallback option now, to be consistent with all the other redis configurations.  I had deliberately left out a default while doing the migration as I didn't want to accidentally send data to the wrong redis (better for it to crash by failing to connect to localhost).

**Pull request merged**